#2081. Fixes active processes and adds all processes filters

ace opened the pull request:
#### :tophat: What? Why?

Fix active processes filter so that it shows only the active ones (currently showing all). Add a new filter for showing all the processes.

#### :pushpin: Related Issues
- Fixes #2081 

#### :clipboard: Subtasks

_None_

### :camera: Screenshots (optional)
<img width="451" alt="processes filter" src="https://user-images.githubusercontent.com/3958/32120070-8d048ea2-bb57-11e7-9c3b-eed4043a8dab.png">

#### :ghost: GIF
![]()

ace commented:
@josepjaume Taking a look at the failing spec, I'm not sure if the `active` detection I've included is right. Can a process have a `nil` `end_date` and be considered `active`?

josepjaume commented:
@ace I think so. There's a bit of overlap here: Participatory processes can have steps, but do have start date and end date as well. This happens because you might not know the end date of a process at first.

While we can fix this by including processes with `nil` end dates as active, we probably should rethink the whole thing.

I'd advocate in favor of removing `start_date` from `participatory_process` and infer that from its steps. We should disallow processes without steps (that's what `assemblies` are for) and create a single, first `step` on process creation.

Steps should only contain `end_date`, preventing time gaps in the middle.

What do you think @carolromero @xabier ?

xabier commented:
hi @josepjaume there can be gaps between phases (e.g. hollidays). Not sure I undertand the problem here. I will check back soon. Sorry.

mrcasals commented:
@josepjaume what should we do with this, then?

josepjaume commented:
@mrcasals this fixes a real bug, one that affects the UX. If @ace doesn't have the time to finish this (totally understandable!), then we probable should take the work over.

ace commented:
@josepjaume I can use some time finishing the fix, just need to know what's the final decision about what to do with the dates.

mrcasals commented:
Hi @ace! Sorry for the trouble. What @josepjaume suggests  lies out of this PR in my opinion and calls for some debate, so let's just focus on fixing the current issue without rethinking how processes work.

I'd consider a process `active` if `start_date` is in the past and `end_date` is either set and in the future or not set, as you suggest in the first comment.

Can you do that please? Also, please add a CHANGELOG entry under `UNRELEASED -> Fixed` following the style of existing entries please 😄 

Thanks!

mrcasals commented:
Oh, please @ace rebase this branch against master so we can ensure tests pass correctly, since we have changed the CI infrastructure quite dramatically over the last weeks.

ace commented:
@mrcasals Fixed the active processes detection, added the CHANGELOG entry and rebased. Fingers crossed for tests, I haven't been able to run them locally (something about a connection pool not found, I'll try to check later).

mrcasals commented:
Wohoo! Thanks for your work @ace, you're getting your first PR merged! 😁

ace commented:
Yeah!! :)
